corehq/apps/auditcare/models.py

```diff
@@ -171,22 +183,19 @@ def audit_view(cls, request, user, view_func, view_kwargs, extra={}):
 )
 
 
+@foreign_value_init
 class AccessAudit(AuditEvent):
-    access_type = StringProperty(choices=ACCESS_CHOICES)
-    ip_address = StringProperty()
-    session_key = StringProperty()  # the django auth session key
-
-    user_agent = StringProperty()
-
-    get_data = StringListProperty()
-    post_data = StringListProperty()
-    http_accept = StringProperty()
-    path_info = StringProperty()
-
-    failures_since_start = IntegerProperty()
-
-    class Meta(object):
-        app_label = 'auditcare'
+    access_type = models.CharField(max_length=16, choices=ACCESS_CHOICES)
+    ip_address = models.CharField(max_length=45, blank=True, default='')
+    session_key = models.CharField(max_length=255, blank=True, null=True)
+    user_agent_fk = models.ForeignKey(
+        UserAgent, null=True, db_index=False, on_delete=models.PROTECT)
+    user_agent = ForeignValue(user_agent_fk, truncate=True)
+    http_accept_fk = models.ForeignKey(
+        HttpAccept, null=True, db_index=False, on_delete=models.PROTECT)
+    http_accept = ForeignValue(http_accept_fk, truncate=True)
+    path_info = models.CharField(max_length=255, blank=True, default='')
+    failures_since_start = models.SmallIntegerField(null=True)
```

Comment:
For other couch to sql migrations I've seen we retain a Couch model and a SQL model until all the old docs are migrated. Presumably when we migrate the data having a schema for the old data will be helpful for converting it to the new format. Do you have a different approach in mind for how we'll do that?

Reply:
I was planning on writing the migration against JSON read directly from Couch since there were no `wrap()` (data migration) methods on these models. I think it will be faster and pretty easy given the simplicity of these models. If they turn out to be useful, the old models can easily be revived from git history.

Reply:
OK cool, thanks for spelling that out for me, since it's a little different form what we often do. The reasoning makes sense, and I don't see any issues with it, and looking over the code I can see why not having to keep both models around makes the change simpler.